File: backend/services/payment/internal/entity/payout.go
package entity

import (
	"encoding/json"
	"time"
)

// PayoutStatus tracks settlement lifecycle.
type PayoutStatus string

const (
	PayoutPending  PayoutStatus = "PENDING"
	PayoutSettled  PayoutStatus = "SETTLED"
)

// PayoutBatch is a store settlement batch created by an admin.
// Executing the batch debits STORE_PAYABLE and marks it SETTLED in one transaction.
type PayoutBatch struct {
	ID          string          `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	StoreID     string          `gorm:"type:uuid;not null"`
	PeriodFrom  time.Time       `gorm:"type:date;not null"`
	PeriodTo    time.Time       `gorm:"type:date;not null"`
	OrderIDs    json.RawMessage `gorm:"type:jsonb;not null;default:'[]'"`
	TotalAmount int64           `gorm:"not null;default:0"`
	Status      PayoutStatus    `gorm:"type:varchar(20);not null;default:'PENDING'"`
	SettledBy   *string         `gorm:"type:uuid"`
	SettledAt   *time.Time      `gorm:"type:timestamptz"`
	CreatedAt   time.Time       `gorm:"type:timestamptz;not null;default:now()"`
}

func (PayoutBatch) TableName() string { return "payout_batches" }

File: frontend/src/features/payouts/types/payout.ts
// Mirrors backend payment-service payout entities.
// Response fields are PascalCase (Go JSON default). Request bodies use snake_case.

export type PayoutStatus = 'PENDING' | 'SETTLED';

// GET /api/v1/admin/settlements?store_id=<uuid>
export interface SettlementPreview {
  StoreID: string;
  PayableBalance: number;
  Orders: SettlementOrder[];
  Total: number;
}

export interface SettlementOrder {
  OrderID: string;
  Amount: number;
  CreatedAt: string;
}

// One payout batch from GET /api/v1/admin/payouts?store_id=...
export interface PayoutBatch {
  ID: string;
  StoreID: string;
  PeriodFrom: string;
  PeriodTo: string;
  OrderIDs: string[];
  TotalAmount: number;
  Status: PayoutStatus;
  SettledBy?: string;
  SettledAt?: string;
  CreatedAt: string;
}

// POST /api/v1/admin/payouts request body
export interface CreatePayoutBody {
  store_id: string;
  period_from: string;
  period_to: string;
  order_ids: string[];
  total_amount: number;
}
